cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/TransformRunner.java

```diff
@@ -94,6 +96,17 @@ public TransformRunner(MapReduceTaskContext<Object, Object> context,
         phase = phase.subsetFrom(ImmutableSet.of(aggregatorName));
       }
     }
+
+    if (!joiners.isEmpty()) {
+      String joinerName = joiners.iterator().next().getName();
```

Comment:
duplication can be removed by doing a union of aggregators and joiners and checking that set.  Because of how the planner works, you're guaranteed that there is at most one joiner or aggregator.